### PR TITLE
Support AMMV2 protocol

### PR DESCRIPTION
**Pull request by BenjaminLu**
Convert AMMV2 `PriceResult` to `Order`. Introduce new `payload` field in AMMV2 protocol.

Run test:

```bash
npm test
```

Should `should signed ammv2 order by uniswap v2` in `new_order.spec.ts`. 

Screen Shot 2021-06-17 at 5 41 53 PM

### CODE REVIEW

File: src/signer/ammv2.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+export const buildSignedOrder = (order, payload, makerAddress, wethAddress): any => {
+  const makerAssetAddress = order.makerAssetAddress.toLowerCase()
+  const takerAssetAddress = order.takerAssetAddress.toLowerCase()
+  // = Rewrite order fields
+  // 1. change maker address to LP pool address
+  order.makerAddress = makerAddress
+  // 2. convert weth to eth
+  if (makerAssetAddress === wethAddress.toLowerCase()) {
+    order.makerAssetData = assetDataUtils.encodeERC20AssetData(NULL_ADDRESS)
+  } else {
```
Comment (tailingchen) [inline src/signer/ammv2.ts:16]: Should overwrite makerAssetAddress and takerAssetAddress here.
Already fixed the bug for AMMV1, please find https://github.com/consenlabs/tokenlon-mmsk/pull/23/commits/995f158b2c724335f88ab422a83272e0d067d152#diff-53fe56f13488d83999f9e39bbbeaf0faab6702e3aa4176aa6215ea679935a27aR15

Comment (BenjaminLu) [reply src/signer/ammv2.ts:17]: Fixed

Comment (BenjaminLu) [reply src/signer/ammv2.ts:17]: Fixed

File: src/signer/ammv2.ts [hunk 2]
```diff
@@ -12,12 +12,14 @@ export const buildSignedOrder = (order, payload, makerAddress, wethAddress): any
   order.makerAddress = makerAddress
   // 2. convert weth to eth
   if (makerAssetAddress === wethAddress.toLowerCase()) {
+    order.makerAssetAddress = NULL_ADDRESS
     order.makerAssetData = assetDataUtils.encodeERC20AssetData(NULL_ADDRESS)
   } else {
```
Comment (tailingchen) [inline src/signer/ammv2.ts:17]: remove else

File: src/signer/ammv2.ts [hunk 3]
```diff
@@ -12,12 +12,14 @@ export const buildSignedOrder = (order, payload, makerAddress, wethAddress): any
...
     order.makerAssetData = assetDataUtils.encodeERC20AssetData(makerAssetAddress)
   }
 
   if (takerAssetAddress === wethAddress.toLowerCase()) {
+    order.takerAssetAddress = NULL_ADDRESS
     order.takerAssetData = assetDataUtils.encodeERC20AssetData(NULL_ADDRESS)
   } else {
```
Comment (tailingchen) [inline src/signer/ammv2.ts:24]: remove else

File: src/signer/ammv2.ts (hunk 2 above)
Comment (BenjaminLu) [reply src/signer/ammv2.ts]: done

File: src/signer/ammv2.ts (hunk 3 above)
Comment (BenjaminLu) [reply src/signer/ammv2.ts]: done

**Review by tailingchen — APPROVED** [review]

**Pull request merged**